skopt/space.py

```diff
@@ -255,7 +258,8 @@ def rvs(self, n_samples=None, random_state=None):
 
     @property
     def transformed_size(self):
-        return len(self.categories)
+        size = len(self.categories)
+        return size if size != 2 else 1
```

Comment:
this is a completely unrelated fix. see https://github.com/scikit-learn/scikit-learn/blob/51a765a/sklearn/preprocessing/label.py#L423 scikit's `LabelBinarizer` outputs arrays of `size == len(categories` for all cases other than passing in exactly two categories to `fit` in which case it outputs a single binary vector